code/game/objects/items.dm

```diff
@@ -983,3 +1000,108 @@ modules/mob/living/carbon/human/life.dm if you die, you will be zoomed out.
 	else
 		active = !active
 	SEND_SIGNAL(src, COMSIG_ITEM_TOGGLE_ACTIVE, active)
+
+///Generates worn icon for sprites on-mob.
+/obj/item/proc/make_worn_icon(body_type, slot_name, inhands, default_icon, default_layer) 
+	//Get the required information about the base icon
+	var/icon/icon2use = get_worn_icon_file(body_type = body_type, slot_name = slot_name, default_icon = default_icon, inhands = inhands)
+	var/state2use = get_worn_icon_state(slot_name = slot_name, inhands = inhands)
+	var/layer2use = get_worn_layer(default_layer = default_layer)
+
+	//Snowflakey inhand icons in a specific slot
+	if(inhands && icon2use == icon_override)
+		switch(slot_name)
+			if(slot_r_hand_str)
+				state2use += "_r"
+			if(slot_l_hand_str)
+				state2use += "_l"
+
+	//testing("[src] (\ref[src]) - Slot: [slot_name], Inhands: [inhands], Worn Icon:[icon2use], Worn State:[state2use], Worn Layer:[layer2use]")
+
+	//Generate the base onmob icon
+	var/icon/standing_icon = icon(icon = icon2use, icon_state = state2use)
+
+	var/image/standing = image(standing_icon)
```

Comment:
Why does this generate an icon which is passed to the image

Reply:
Now seeing, can probably do a direct image, it used to be an icon since apply_custom() was done on icon and had an addblends, but now that ive got rid of those can make direct icons